server/commands/teamUpdater.ts

```diff
@@ -0,0 +1,73 @@
+import { Event, Team, User } from "@server/models";
+
+type TeamUpdaterProps = {
+  params: Partial<Team>;
+  ip?: string;
+  user: User;
+  team: Team;
+};
+
+const teamUpdater = async ({ params, user, team, ip }: TeamUpdaterProps) => {
+  const {
+    name,
+    avatarUrl,
+    subdomain,
+    sharing,
+    guestSignin,
+    documentEmbeds,
+    collaborativeEditing,
+    defaultCollectionId,
+    defaultUserRole,
+  } = params;
+
+  if (subdomain !== undefined && process.env.SUBDOMAINS_ENABLED === "true") {
+    team.subdomain = subdomain === "" ? null : subdomain;
+  }
+
+  if (name) {
+    team.name = name;
+  }
+  if (sharing !== undefined) {
+    team.sharing = sharing;
+  }
+  if (documentEmbeds !== undefined) {
+    team.documentEmbeds = documentEmbeds;
+  }
+  if (guestSignin !== undefined) {
+    team.guestSignin = guestSignin;
+  }
+  if (avatarUrl !== undefined) {
+    team.avatarUrl = avatarUrl;
+  }
+  if (defaultCollectionId !== undefined) {
+    team.defaultCollectionId = defaultCollectionId;
+  }
+  if (collaborativeEditing !== undefined) {
+    team.collaborativeEditing = collaborativeEditing;
+  }
+  if (defaultUserRole !== undefined) {
+    team.defaultUserRole = defaultUserRole;
+  }
+
+  const changes = team.changed();
+
+  const savedTeam = await team.save();
```

Comment:
Could you wrap this `save` and `Event.create` in a transaction? (see other commands) – trying to make sure that we never have a model update without the associated event log in commands – gradual best practice…